eng/scripts/MgmtTestLib.ps1

```diff
@@ -0,0 +1,350 @@
+        if ($LASTEXITCODE)
+        {
+            Write-Host "##[error]Error running autorest.gotest"
+            exit $LASTEXITCODE
+        }
+    }
+
+    if ($format)
+    {
+        Write-Host "##[command]Executing gofmt -s -w . in " $sdkDirectory
+        gofmt -s -w .
+        if ($LASTEXITCODE) { exit $LASTEXITCODE }
+    }
+
+    if ($tidy)
+    {
+        Write-Host "##[command]Executing go mod tidy in " $sdkDirectory
+        go mod tidy
+        if ($LASTEXITCODE) { exit $LASTEXITCODE }
+    }
+
+    if (!$skipBuild)
+    {
+        Write-Host "##[command]Executing go build -x -v ./... in " $sdkDirectory
+        go build -x -v ./...
+        Write-Host "##[command]Build Complete!"
+        if ($LASTEXITCODE) { exit $LASTEXITCODE }
+    }
+
+    if ($vet)
+    {
+        Write-Host "##[command]Executing go vet ./... in " $sdkDirectory
+        go vet ./...
+    }
+}
+
+. (Join-Path $PSScriptRoot .. common scripts common.ps1)
+
+function PrepareMockServer()
+{
+    param(
+        [string]$MOCK_SERVER_NAME = "mock-server"
+    )
+    # install mock server
+    $folder = Join-Path $env:TEMP "mock-service-host"
+    StopMockServer -MOCK_SERVER_NAME $MOCK_SERVER_NAME
+    try
+    {
+        Remove-Item -Recurse -Force -Path $folder
+    }
+    catch
+    {
+        Write-Host "Mock service host folder: $folder not existed"
+    }
+    New-Item -ItemType Directory -Force -Path $folder
+    Set-Location $folder
+    npm install @azure-tools/mock-service-host
+}
+
+function StartMockServer()
+{
+    param(
+        [string]$MOCK_SERVER_NAME = "mock-server",
+        [string]$MOCK_SERVER_READY = "validator initialized",
+        [string]$MOCK_SERVER_WAIT_TIME = 600,
+        [string]$specDir = "",
+        [string]$rpSDKFolder,
+        [string]$AUTOREST_CONFIG_FILE = "autorest.md"
+    )
+    $folder = Join-Path $env:TEMP  "mock-service-host"
+    Set-Location $folder
+
+    # change .env file to use the specific swagger file
+    $envFile = Join-Path $folder .env
+    if ([string]::IsNullOrEmpty($rpSDKFolder))
+    {
+        if ([string]::IsNullOrEmpty($specDir)) {
+            $swaggerInfo = [PSCustomObject]@{
+                isRepoUrl = $true
+                path = "https://github.com/Azure/azure-rest-api-specs"
+                specName = "*"
+                org = "Azure"
+                branch = "main"
+                commitID = ""
+            }
+        } else {
+            $swaggerInfo = [PSCustomObject]@{
+                isRepoUrl = $false
+                path = $specDir
+                specName = "*"
+                org = ""
+                branch = ""
+                commitID = ""
+            }
+        }
+        
+    } else {
+        Write-Host "Retrieve Swagger info"
+        $swaggerInfo = GetSwaggerInfo -dir $rpSDKFolder -AUTOREST_CONFIG_FILE $AUTOREST_CONFIG_FILE
+    }
```

Comment:
All the logic is to get swagger info from some param. Can we move it all to the `GetSwaggerInfo` func and just pass `SwaggerInfo` object to this func?